src/js/player.js

```diff
@@ -1347,39 +1349,32 @@ class Player extends Component {
 
       this.muted(true);
 
-      const playPromise = this.play();
+      const restoreMuted = () => {
+        this.off('play-terminated', restoreMuted);
+        this.muted(previouslyMuted);
+      };
 
-      if (!playPromise || !playPromise.then || !playPromise.catch) {
-        return;
-      }
+      // restore muted on play-terminated as well
+      this.one('play-terminated', restoreMuted);
 
-      return playPromise.catch((e) => {
-        // restore old value of muted on failure
-        this.muted(previouslyMuted);
-      });
+      return this.play().then(() => {
+        this.off('play-terminated', restoreMuted);
+      }).catch(restoreMuted);
     };
 
     let promise;
 
-    if (type === 'any') {
-      promise = this.play();
-
-      if (promise && promise.then && promise.catch) {
-        promise.catch(() => {
-          return muted();
-        });
-      }
-    } else if (type === 'muted') {
+    // if muted defaults to true
+    // the only thing we can do is call play
+    if (type === 'any' && this.muted() !== true) {
+      promise = this.play().catch(muted);
+    } else if (type === 'muted' && this.muted() !== true) {
       promise = muted();
     } else {
       promise = this.play();
     }
 
-    if (!promise || !promise.then || !promise.catch) {
-      return;
-    }
-
-    return promise.then(() => {
+    promise.then(() => {
```

Comment:
is the return value from `manualAutoplay_` not used anywhere?

Reply:
It isn't used anywhere, but we should probably still return it